lib/AST/DiagnosticEngine.cpp

```diff
@@ -311,6 +358,29 @@ void Diagnostic::addChildNote(Diagnostic &&D) {
   ChildNotes.push_back(std::move(D));
 }
 
+LocalizationProducer::LocalizationProducer(std::string locale,
+                                           std::string path) {
```

Comment:
`LocalizationProducer` shouldn't know anything about a concrete implementation, this should be a constructor of `YAMLLocalizationProducer`.